mlrun/db/base.py

```diff
@@ -142,6 +143,85 @@ def delete_function(self, name: str, project: str = ""):
     def list_functions(self, name=None, project="", tag="", labels=None):
         pass
 
+    @abstractmethod
+    def tag_objects(
+        self,
+        project: str,
+        tag_name: str,
+        tag_objects: schemas.TagObjects,
+        replace: bool = False,
+    ):
+        pass
+
+    def delete_objects_tag(
+        self,
+        project: str,
+        tag_name: str,
+        tag_objects: schemas.TagObjects,
+    ):
+        pass
+
+    def tag_artifacts(
+        self,
+        artifacts,
+        project: str,
+        tag_name: str,
+        replace: bool = False,
+    ):
+        pass
+
+    def delete_artifacts_tags(
+        self,
+        artifacts,
+        project: str,
+        tag_name: str,
+    ):
+        pass
+
+    @staticmethod
+    def _resolve_artifacts_to_tag_objects(
+        artifacts,
+    ) -> schemas.TagObjects:
+        """
+        :param artifacts: Can be a list of :py:class:`~mlrun.artifacts.Artifact` objects or
+            dictionaries, or a single object.
+        :return: :py:class:`~mlrun.api.schemas.TagObjects`
+        """
+        from mlrun.artifacts.base import Artifact
```

Comment:
Please either move the import to the top of the file or add a comment as to why you need the import here.
(maybe also convert to a `import mlrun.artifacts.base as artifacts_base` and then use `artifacts_base.Artifact` to avoid import loops)